packages/algob/src/lib/files.ts

```diff
@@ -64,3 +72,27 @@ export function loadFromYamlFileSilentWithMessage (
     return defaultYamlValue(options);
   }
 }
+
+export async function readMsigFromFile (filename: string): Promise<Object> {
+  if (!filename.endsWith(msigExt)) {
+    throw new Error(`filename "${filename}" must end with "${msigExt}"`);
+  }
+  try {
+    const p = path.join(ASSETS_DIR, filename);
+
+    // Extracting teal code from .mlsig and dumping to .teal to get logic signature
+    const tealFile = filename.split(msigExt)[0] + tealExt;
+    const tealPath = path.join(ASSETS_DIR, tealFile); // assets/<file_name>.teal
+    const tealCode = fs.readFileSync(p, 'utf8').split("LogicSig: ")[0];
+
+    // Write logic code to file with .teal ext
+    writeFile(tealPath, tealCode);
```

Comment:
why do we write the extracted TEAL code to a new file?

If you want to use the compile functionality, then you can use a `CompileOp` and add a new function: `ensureCompiledCode` which can use  part of  `ensureCompiled` (if that is the case then we should extract that common code from `ensureCompiled` and use it in both functions).

Reply:
Continuing on discord.